skops/cli/tests/test_convert.py

```diff
@@ -0,0 +1,138 @@
+import logging
+import pathlib
+import pickle as pkl
+from unittest import mock
+
+import numpy as np
+import pytest
+
+from skops.cli import _convert
+from skops.io import load
+
+
+class MockUnsafeType:
+    def __init__(self):
+        pass
+
+
+class TestConvert:
+    model_name = "some_model_name"
+
+    @pytest.fixture
+    def safe_obj(self):
+        return np.ndarray([1, 2, 3, 4])
+
+    @pytest.fixture
+    def unsafe_obj(self):
+        return MockUnsafeType()
+
+    @pytest.fixture
+    def pkl_path(self, tmp_path):
+        return tmp_path / f"{self.model_name}.pkl"
+
+    @pytest.fixture
+    def skops_path(self, tmp_path):
+        return tmp_path / f"{self.model_name}.skops"
+
+    @pytest.fixture
+    def write_safe_file(self, pkl_path, safe_obj):
+        with open(pkl_path, "wb") as f:
+            pkl.dump(safe_obj, f)
+
+    @pytest.fixture
+    def write_unsafe_file(self, pkl_path, unsafe_obj):
+        with open(pkl_path, "wb") as f:
+            pkl.dump(unsafe_obj, f)
+
+    def test_base_case_works_as_expected(
+        self, pkl_path, tmp_path, skops_path, write_safe_file, safe_obj, caplog
+    ):
+        _convert._convert_file(pkl_path, skops_path)
+        persisted_obj = load(skops_path)
+        assert np.array_equal(persisted_obj, safe_obj)
+        assert MockUnsafeType.__name__ not in caplog.text
```

Comment:
I'm a bit worried about this specific test, because it can easily pass because of some change in the default log level. Not sure what the best way would be to avoid this, but one way would be to use dependency injection. That would mean that we pass a `logger` object to `_convert_file`, which by default is `logging.Logger()`, but in the test can be, e.g., a mock. Then we can run this check on the mock logger, instead of relying on `caplog`.

Reply:
Happy to rework to use dependency injection, I think it makes sense to keep things more testable.

Reply:
Talked to a friend about using DI in this case. Although I think it's not a bad use of DI, it might be a bit unorthodox when it comes to logging, so if you feel this makes the code harder to maintain, feel free to ignore the suggestion.